### PR TITLE
refactor persistence plugin

### PR DESCRIPTION
**Pull request by titovmx**

### CODE REVIEW

File: plugin/persistence/persistence.view.js [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+		this.state = {
+			editItem: null,
+			oldValue: null
+		};
+		this.model = model;
+		this.storageKey = `q-grid:${model.grid().id}:${model.persistence().id}:persistence-list`;
+		this.persistenceService = new PersistenceService(model);
+		this.submitEvent = new Event();
+		this.cancelEvent = new Event();
+		this.closeEvent = new Event();
```
Comment (klumba12) [inline plugin/persistence/persistence.view.js]: lets use ony cancelEvent, to close or not close decides plugin itself

File: src/plugins/persistence/persistence.js [hunk 2]
```diff
@@ -48,14 +37,13 @@ class Persistence extends Plugin {
 
 						const config = {
 							attachTo: angular.element(this.$document[0].body), // eslint-disable-line no-undef
-							controller: PersistencePanelController,
+							controller: () => {},
```
Comment (klumba12) [inline src/plugins/persistence/persistence.js]: can we just use persistenceView in controller and inside of controller make $scope.persistence = new PersView()?
so no need to write $persistence.view?

File: plugin/persistence/persistence.view.js [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+	isActive(item) {
+		return JSON.stringify(item.model) === JSON.stringify(this.persistenceService.save()); // eslint-disable-line angular/json-functions
+	}
+
+	stringify(item) {
+		const model = item.model;
+		const targets = [];
+		const settings = this.model.persistence().settings;
+
+		for (let key in settings) {
```
Comment (etokareva) [inline plugin/persistence/persistence.view.js:170]: do we want to check all settings properties there or own?

Comment (titovmx) [reply plugin/persistence/persistence.view.js:170]: all iterable properties exactly

**Review by klumba12 — APPROVED** [review]

**Pull request merged**